src/chk/chk_internal.h

```diff
@@ -475,4 +493,58 @@ void chk_vos_init(void);
 
 void chk_vos_fini(void);
 
+static inline bool
+chk_rank_in_list(d_rank_list_t *rlist, d_rank_t rank)
```

Comment:
[Nit] Just FYI, there's d_rank_list_find.

Reply:
Similar, the function chk_rank_in_list() will be enhanced via sort the ranks.